src/Tribe/Commerce/Cart.php

```diff
@@ -34,15 +34,67 @@ public function process_cart() {
 				$data = stripslashes( $data );
 				$data = json_decode( $data, true );
 			}
+
+			$data = array_merge( $_POST, $data );
 		}
 
-		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? $data['tribe_tickets_post_id'] : null;
-		$provider = isset( $data['tribe_tickets_provider'] ) ? $data['tribe_tickets_provider'] : null;
+		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? absint( $data['tribe_tickets_post_id'] ) : null;
+		$provider = isset( $data['tribe_tickets_provider'] ) ? sanitize_text_field( $data['tribe_tickets_provider'] ) : null;
 		$tickets  = isset( $data['tribe_tickets_tickets'] ) ? $data['tribe_tickets_tickets'] : null;
 		$meta     = isset( $data['tribe_tickets_meta'] ) ? $data['tribe_tickets_meta'] : null;
 
+		if ( null === $provider ) {
+			if ( ! empty( $_GET['provider'] ) ) {
+				$provider = sanitize_text_field( $_GET['provider'] );
+			} elseif ( ! empty( $_POST['provider'] ) ) {
+				$provider = sanitize_text_field( $_POST['provider'] );
+			}
+		}
+
+		$tribe_commerce_providers = [
+			'tpp',
+			'tribe-commerce',
+			'tribe_tpp_attendees',
+			'Tribe__Tickets__Commerce__PayPal__Main',
+		];
+
+		$is_tribe_commerce = in_array( $provider, $tribe_commerce_providers, true );
+
+		// Simplify provider for Tribe Commerce.
+		if ( $is_tribe_commerce ) {
+			$provider = 'tribe-commerce';
+		}
+
 		// On AR Page, we use replace logic, not additive.
 		$is_ar_modal = empty( $_POST['tribe_tickets_ar_page'] );
+		$additive    = $is_ar_modal && ! $is_tribe_commerce;
+
+		if ( $is_tribe_commerce ) {
+			if ( null === $post_id ) {
+				if ( ! empty( $_GET['tribe_tickets_post_id'] ) ) {
+					// Get post ID from current URL parameter.
+					$post_id = absint( $_GET['tribe_tickets_post_id'] );
+				} else {
+					// Detect post ID for Tribe Commerce from the first ticket (no cart, we only support one post at a time).
```

Comment:
Detect post ID from the first ticket in the cart if it's not submitted with POST